cmd/oci-trace-hook/trace.go

```diff
@@ -0,0 +1,276 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/binary"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	bpf "github.com/iovisor/gobpf/bcc"
+	spec "github.com/opencontainers/runtime-spec/specs-go"
+	seccomp "github.com/seccomp/libseccomp-golang"
+	"github.com/sirupsen/logrus"
+)
+
+// event
+type event struct {
+	Pid uint32
+	ID  uint32
+	// Inum    uint
+	Command [16]byte
+}
+
+// calls: used to store all the syscall name and the number of times it's been called
+type calls map[string]int
+
+const source string = `
+#include <linux/bpf.h>
+#include <linux/nsproxy.h>
+#include <linux/pid_namespace.h>
+#include <linux/ns_common.h>
+#include <linux/sched.h>
+#include <linux/tracepoint.h>
+
+BPF_HASH(parent_namespace, u64, unsigned int);
+BPF_PERF_OUTPUT(events);
+
+struct data_t
+{
+    u32 pid;
+    u32 id;
+    char comm[16];
+};
+
+// enter_trace watches all of the syscalls made to the kernel by watching,
+// raw_syscalls:sys_enter tracepoint, it then logs all the syscalls made by 
+// processes inside certain pid namespace
+int enter_trace(struct tracepoint__raw_syscalls__sys_enter *args)
+{
+    struct data_t data = {};
+    u64 key = 0;
+    unsigned int zero = 0;
+    struct task_struct *task;
+
+    data.pid = bpf_get_current_pid_tgid();
+    data.id = (int)args->id;
+    bpf_get_current_comm(&data.comm, sizeof(data.comm));
+
+    task = (struct task_struct *)bpf_get_current_task();
+    struct nsproxy *ns = task->nsproxy;
+    unsigned int inum = ns->pid_ns_for_children->ns.inum;
+
+    if (data.pid == $PARENT_PID)
+    {
+        parent_namespace.update(&key, &inum);
+    }
+    unsigned int *parent_inum = parent_namespace.lookup_or_init(&key, &zero);
+
+    if (*parent_inum != inum)
+    {
+        return 0;
+    }
+
+    events.perf_submit(args, &data, sizeof(data));
+    return 0;
+}
+`
+
+func main() {
+
+	terminate := flag.Bool("t", false, "send SIGINT to floating process")
+	runBPF := flag.Int("r", 0, "-r [PID] run the BPF function and attach to the pid")
+	fileName := flag.String("f", "profile.json", "path of the file to save the seccomp profile")
```

Comment:
We need to make it an absolute path, for instance, with [filepath.Abs()](https://golang.org/pkg/path/filepath/#Abs).

Reply:
This is still valid. If I run `podman run --generate-secomp myprofile.json alpine ls` I want `myprofile.json` to show up in my working directory. So we need to do a:
```golang
fileName, err = filepath.Abs(fileName)
if err ...
```
^ to be safe, we should do the same for the `generate-seccomp` path.